auth/detect/detect.go

```diff
@@ -96,6 +116,30 @@ func (c *Credentials) UniverseDomain() string {
 	return c.universeDomain
 }
 
+// ValidateUniverseDomain verifies TODO(chridsmith): doc
+func (c *Credentials) ValidateUniverseDomain(ctx context.Context, clientUniverseDomain string) error {
```

Comment:
Lets move this closer to the transport layers

Reply:
Done, thanks!